druid-shell/src/backend/wayland/application.rs

```diff
@@ -180,120 +167,136 @@ impl Application {
-                        version,
-                    } => {
-                        if interface.as_str() == "wl_output" && version >= 3 {
-                            let output = registry.bind::<WlOutput>(3, id);
-                            let output = Output::new(id, output);
-                            let oid = output.id();
-                            let gid = output.gid;
-                            let previous = weak_outputs
-                                .upgrade()
-                                .unwrap()
-                                .borrow_mut()
-                                .insert(oid, output.clone());
-                            assert!(
-                                previous.is_none(),
-                                "internal: wayland should always use new IDs"
-                            );
-                            tracing::trace!("output added {:?} {:?}", gid, oid);
-                            output.wl_output.quick_assign(with_cloned!(weak_outputs, gid, oid, outputsaddedtx; move |a, event, b| {
-                                    tracing::trace!("output event {:?} {:?} {:?}", a, event, b);
-                                    match weak_outputs.upgrade().unwrap().borrow_mut().get_mut(&oid) {
-                                        Some(o) => o.process_event(event, &outputsaddedtx),
-                                        None => tracing::warn!(
-                                            "wayland sent an event for an output that doesn't exist global({:?}) proxy({:?}) {:?}",
-                                            &gid,
-                                            &oid,
-                                            &event,
-                                        ),
-                                    }
-                                }));
-                        } else if interface.as_str() == "wl_seat" && version >= 7 {
-                            let new_seat = registry.bind::<WlSeat>(7, id);
-                            let prev_seat = weak_seats
-                                .upgrade()
-                                .unwrap()
-                                .borrow_mut()
-                                .insert(id, Rc::new(RefCell::new(Seat::new(new_seat))));
-                            assert!(
-                                prev_seat.is_none(),
-                                "internal: wayland should always use new IDs"
-                            );
-                            // Defer setting up the pointer/keyboard event handling until we've
-                            // finished constructing the `Application`. That way we can pass it as a
-                            // parameter.
-                        }
+
+        let dispatcher = display::Dispatcher::default();
+
+        display::GlobalEventDispatch::subscribe(&dispatcher, move |event: &'_ wl::GlobalEvent, registry: &'_ wl::Attached<wl_registry::WlRegistry>, _ctx: &'_ wl::DispatchData| {
```

Comment:
looking to pull this entire logic into the outputs module in next PR where I'll be fixing a layershell reinitialization bug.